### DISCUSSION TITLE
[Idea] Introduce layout field for building complex row/columns layouts

### DISCUSSION THREAD

**Discussion by wsydney76** [craftcms/cms#8998] [Ideas]
### Description

Not a concrete FR, just an idea for upcoming versions...

Still struggling to build a user friendly UI for building these typical page-builder like layouts (rows with different number/widths of columns, background colors etc.) with every imaginable combination of Matrix/NEO/Super Table, so I'd like to point out an idea <s>stolen from</s> inspired by [Kirby 3.5](https://getkirby.com/docs/reference/panel/fields/layout) (and mayby others I dont't know about):

This field allows to visually compose/edit a page by adding rows with predefined columns layouts (eg. 1/3 - 2/3), adding settings to it (e.g. background colors, custom classes) and filling the columns with blocks from an associated matrix field.

That matrix field in turn could only contain 'real' content without getting messed up with layout stuff.

Seems to me that this would'nt even change the way matrix content is stored, the layout field just has to keep track of which blocks are positioned in which column.

**Comment by brandonkelly** [comment]
Good idea, maybe worth us thinking about it. Check out the [Grid](https://plugins.craftcms.com/grid) plugin in the meantime ;)

**Comment by piotrpog** [comment]
Juast adding my two cents - i think that [Article](https://imperavi.com/article/) script created by same people that made Redactor, could be useful for such things.

**Comment by yellostudio** [comment]
Hopefully this is still in mind to sort...
